src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -79,45 +73,75 @@ export class HumanDescriptionMapper {
     }
   }
 
-  createHumanDescription(
-    descriptionFragments: HumanDescriptionFragment[],
-    token: Token | null,
-  ): string {
-    return descriptionFragments
-      .map((fragment) => {
+  async enrichFragments(
+    fragments: HumanDescriptionFragment[],
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichHumanDescriptionFragment[]> {
+    return Promise.all(
+      fragments.map(async (fragment) => {
         switch (fragment.type) {
           case ValueType.TokenValue:
-            if (!token?.decimals) return fragment.value.amount;
-
-            // Unlimited approval
-            if (fragment.value.amount === MAX_UINT256) {
-              return `unlimited ${token.symbol}`;
-            }
-
-            return `${formatUnits(fragment.value.amount, token.decimals)} ${
-              token.symbol
-            }`;
+            return this.enrichTokenValue(fragment, transaction, chainId);
           case ValueType.Address:
-            return this.shortenAddress(fragment.value);
           default:
-            return fragment.value;
+            return fragment;
         }
-      })
-      .join(' ');
+      }),
+    );
   }
 
-  shortenAddress(address: Hex, length = 4): string {
-    if (!isAddress(address)) {
-      throw Error('Invalid address');
-    }
+  async enrichTokenValue(
+    fragment: TokenValueFragment,
+    transaction: MultisigTransaction | ModuleTransaction,
+    chainId: string,
+  ): Promise<RichTokenValueFragment> {
+    try {
+      const token = await this.tokenRepository.getToken({
+        chainId,
+        address: transaction.to,
+      });
 
-    const visibleCharactersLength = length * 2 + 2;
+      return <RichTokenValueFragment>{
+        type: ValueType.TokenValue,
+        value: {
+          token,
+          amount:
+            fragment.value.amount === MAX_UINT256
+              ? 'unlimited'
+              : token.decimals
+              ? formatUnits(fragment.value.amount, token.decimals)
+              : fragment.value.amount.toString(),
```

Comment:
I think the chain of conditions for the `amount` can be confusing for the reader.

Suggestion (extract the conditions):

```typescript
let amountLabel: string;
if (fragment.value.amount === MAX_UINT256) {
  amountLabel = 'unlimited';
} else if (token && token.decimals) {
  amountLabel = formatUnits(fragment.value.amount, token.decimals);
} else {
  amountLabel = fragment.value.amount.toString();
}
```